src/EventListener/WriteListener.php

```diff
@@ -59,7 +59,8 @@ public function onKernelView(GetResponseForControllerResultEvent $event)
                     @trigger_error(sprintf('Returning void from %s::persist() is deprecated since API Platform 2.3 and will not be supported in API Platform 3, an object should always be returned.', DataPersisterInterface::class), E_USER_DEPRECATED);
                 }
 
-                $event->setControllerResult($persistResult ?? $controllerResult);
+                $controllerResult = $persistResult ?? $controllerResult;
+                $event->setControllerResult($controllerResult);
```

Comment:
This change has a open discussion here: https://github.com/api-platform/core/pull/2235#discussion_r223208885 and I believe it breaks a feature (see WriteListenerTest::testOnKernelViewWithControllerResultAndPersistWithImmutableResource, the resource isn't immutable anymore?) ?